src/anonymousUserTracking/anonymousUserEventManager.ts

```diff
@@ -0,0 +1,367 @@
+    }
+  }
+
+  getAnonCriteria() {
+    baseIterableRequest<IterableResponse>({
+      method: 'GET',
+      url: GET_CRITERIA_PATH,
+      data: {},
+      validation: {}
+    })
+      .then((response) => {
+        const criteriaData: any = response.data;
+        if (criteriaData) {
+          localStorage.setItem(
+            SHARED_PREFS_CRITERIA,
+            JSON.stringify(criteriaData)
+          );
+        }
+      })
+      .catch((e) => {
+        console.log('response', e);
+      });
+  }
+
+  async trackAnonEvent(payload: InAppTrackRequestParams) {
+    const newDataObject = {
+      [KEY_EVENT_NAME]: payload.eventName,
+      [KEY_CREATED_AT]: this.getCurrentTime(),
+      [KEY_DATA_FIELDS]: payload.dataFields,
+      [KEY_CREATE_NEW_FIELDS]: true,
+      [SHARED_PREFS_EVENT_TYPE]: TRACK_EVENT
+    };
+    this.storeEventListToLocalStorage(newDataObject, false);
+  }
+
+  async trackAnonUpdateUser(payload: UpdateUserParams) {
+    const newDataObject = {
+      ...payload.dataFields,
+      [SHARED_PREFS_EVENT_TYPE]: UPDATE_USER
+    };
+    this.storeEventListToLocalStorage(newDataObject, true);
+  }
+
+  async trackAnonPurchaseEvent(payload: TrackPurchaseRequestParams) {
+    const newDataObject = {
+      [KEY_ITEMS]: payload.items,
+      [KEY_CREATED_AT]: this.getCurrentTime(),
+      [KEY_DATA_FIELDS]: payload.dataFields,
+      [KEY_TOTAL]: payload.total,
+      [SHARED_PREFS_EVENT_TYPE]: TRACK_PURCHASE
+    };
+    this.storeEventListToLocalStorage(newDataObject, false);
+  }
+
+  async trackAnonUpdateCart(payload: UpdateCartRequestParams) {
+    const newDataObject = {
+      [KEY_ITEMS]: payload.items,
+      [SHARED_PREFS_EVENT_TYPE]: TRACK_UPDATE_CART,
+      [KEY_PREFER_USERID]: true,
+      [KEY_CREATED_AT]: this.getCurrentTime()
+    };
+    this.storeEventListToLocalStorage(newDataObject, false);
+  }
+
+  private async checkCriteriaCompletion() {
+    const criteriaData = localStorage.getItem(SHARED_PREFS_CRITERIA);
+    const localStoredEventList = localStorage.getItem(
+      SHARED_PREFS_EVENT_LIST_KEY
+    );
+    try {
+      if (criteriaData && localStoredEventList) {
+        const checker = new CriteriaCompletionChecker(localStoredEventList);
+        return checker.getMatchedCriteria(criteriaData);
+      }
+    } catch (error) {
+      console.error('checkCriteriaCompletion', error);
+    }
+
+    return null;
+  }
+
+  private async createKnownUser(criteriaId: string) {
+    const userData = localStorage.getItem(SHARED_PREFS_ANON_SESSIONS);
+    const userId = uuidv4();
+
+    if (userData) {
+      const userSessionInfo = JSON.parse(userData);
+      const userDataJson = userSessionInfo[SHARED_PREFS_ANON_SESSIONS];
+      const payload: TrackAnonSessionParams = {
+        user: {
+          userId,
+          mergeNestedObjects: true,
+          createNewFields: true
+        },
+        createdAt: this.getCurrentTime(),
+        deviceInfo: {
+          appPackageName: window.location.hostname,
```

Comment:
`appPackageName` needs to be a constructor on the class so devs can determine it.

Reply:
AnonymousUserEventManager class won't be exposed to developer as all the functions are supposed to be called internally only, so dev would have no way to pass this appPackageName. (unless we make them pass it inside configOptions)

```
const initializeParams: WithJWTParams = {
    authToken: process.env.API_KEY || '',
    configOptions: {
      isEuIterableService: false,
      dangerouslyAllowJsPopups: true
    },
```

Reply:
@bradumbaugh at some point, we should move `appPackageName` to the config here.